packages/core/src/@types/configs.ts

```diff
@@ -51,3 +51,18 @@ export interface ZapConfigArgs extends OrderConfigArgs {
 export interface WithdrawConfigArgs extends OrderConfigArgs {
   suppliedLPAsset: IAsset;
 }
+
+/** A map of pools with their associated weight. */
+export type DelegationProgramPools = Map<string, bigint>;
```

Comment:
why would you expect the `weight` to be a bigint instead of a plain number?

Reply:
It can't be fractional, and it's encoded as a bigint in the datum